x/app-router/keeper/liquidstake.go

```diff
@@ -0,0 +1,84 @@
+package keeper
+
+import (
+	"github.com/armon/go-metrics"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+
+	"github.com/Stride-Labs/stride/v3/x/app-router/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v3/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v3/x/stakeibc/types"
+)
+
+func (k Keeper) TryLiquidStaking(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	newData transfertypes.FungibleTokenPacketData,
+	parsedReceiver *types.ParsedReceiver,
+	ack ibcexported.Acknowledgement,
+) ibcexported.Acknowledgement {
+	params := k.GetParams(ctx)
+	if !params.Active {
+		return channeltypes.NewErrorAcknowledgement("packet forwarding param is not active")
+	}
+	// recalculate denom, skip checks that were already done in app.OnRecvPacket
+	var denom string
+	// In this case, we can't process a liquid staking transaction, because we're dealing with STRD tokens
+	if transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), newData.Denom) {
+		return channeltypes.NewErrorAcknowledgement("not a supported token for liquid staking")
+	} else {
+		prefixedDenom := transfertypes.GetDenomPrefix(packet.GetDestPort(), packet.GetDestChannel()) + newData.Denom
+		denom = transfertypes.ParseDenomTrace(prefixedDenom).BaseDenom
+	}
+	amount, ok := sdk.NewIntFromString(newData.Amount)
+	if !ok {
+		return channeltypes.NewErrorAcknowledgement("not a parsable amount field")
+	}
+	var token = sdk.NewCoin(denom, amount)
+
+	err := k.RunLiquidStake(ctx, parsedReceiver.StrideAccAddress, token, []metrics.Label{})
+	if err != nil {
+		ack = channeltypes.NewErrorAcknowledgement(err.Error())
+	}
+	return ack
+}
+
+func (k Keeper) RunLiquidStake(ctx sdk.Context, addr sdk.AccAddress, token sdk.Coin, labels []metrics.Label) error {
+	msg := &stakeibctypes.MsgLiquidStake{
+		Creator:   addr.String(),
+		Amount:    token.Amount.Uint64(),
+		HostDenom: token.Denom,
+	}
+
+	if err := msg.ValidateBasic(); err != nil {
+		return err
+	}
+
+	msgServer := stakeibckeeper.NewMsgServerImpl(k.stakeibcKeeper)
+	_, err := msgServer.LiquidStake(
+		sdk.WrapSDKContext(ctx),
+		msg,
+	)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInsufficientFunds, err.Error())
+	}
+
+	defer func() {
```

Comment:
What is this code block doing?

Reply:
Removed here https://github.com/Stride-Labs/stride/pull/432/commits/d17ffb6028a32eba576bf75d10c226dfc539278e